### ISSUE TITLE
[Epic]: Migrate from enzyme to testing-library

### ISSUE THREAD

**Issue by ecraig12345** [microsoft/fluentui#21663]
_Part of CDA - internal deadline for this is May 30th 2023._ 

if your package currently uses Enzyme based test, we recommend that your/team perform the following actions.

- Use React Testing Library (RTL) for authoring ‘new’ test cases starting today.
- Convert all existing Enzyme based test to RTL by 31th May 2023.

We don't have resources to make it till that time, but we should be able to target end of Q1/2 2024.

### Background

We'd like to migrate from enzyme to [testing-library](https://testing-library.com/docs/) since enzyme is effectively unmaintained (over a year and no official React 17 adapter) and testing-library provides a better, more reliable testing experience that more closely resembles user interaction.

~~This has become a higher priority because we'd like to upgrade our repo to React 17 (#20145) but it's being blocked by the lack of an Enzyme adapter.~~ EDIT: Between testing-library and cypress migrations, and possibly fixes to the unofficial adapter, the unofficial adapter now works with our remaining tests. So this is no longer a blocker for in-repo React 17 adoption. It [*will* block React 18 adoption](https://dev.to/wojtekmaj/enzyme-is-dead-now-what-ekl) in-repo, but we probably won't be there for a few months at least.

testing-library has an excellent [guide to migrating from enzyme](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/) -- it's a bit different way of thinking rather than a 1-1 mapping that could be automated, but overall a much better approach.

### Steps

- [x] Requested that v8 core team use testing-library for any new tests
- [x] Remove from converged packages
  - this was mostly already done
  - [x] Remove enzyme test setup from converged packages - #22245
- [x] Remove from react-conformance #22009
- [x] Remove from `@fluentui/a11y-testing` - #22262
- [ ] Ban new usage of enzyme APIs with lint rule ~or danger check~
- [ ] Work on migrating v8 component tests to RTL (react testing-library)
  - [x] List of Components that need to be converted #22588
  - Could split this up with people
  - Some tests will be ported to cypress instead as it makes sense #21662
  - [x] react-conformance (#22009)
  - [x] Remove config in places where it's not needed and port a couple simple packages to testing-library (#22437)
- [ ] Remove from v0? (up to n* team)
- [x] Document best practices in wiki: [Wiki](https://github.com/microsoft/fluentui/wiki/Testing-with-Jest#testing-components)
  - link to testing-library docs
  - designate a component for demonstrating testing best practices?
  - don't test every single prop? (perf concerns)

### Related issues

- #20507
- #20145
- #17101

**Comment by sr-shifu** [comment]
Doing deprecation of Enzyme in my project, I figured out there is a way to make this process more manageable and automated: https://thesametech.com/migrate-away-from-enzyme/
Please check out my article, and let me know what you think!
